notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * This file leverages Modal Bootstrap to display a single notice, like an Term of Service
+ */
+
+var upmodal_notice = upmodal_notice || {};
+
+if (!upmodal_notice.init) {
+
+  upmodal_notice.init = true;
+
+  (function() {
+
+    var defaults = {
+      selectors: {
+        content:         '.np-content',
+        title:           '.np-title',
+        form:            '.np-action-form',
+        body:            '.np-body',
+        link:            '.np-link',
+        actions:         '.np-actions',
+        actionTemplate:  '.np-action-template',
+        closeButton:     '.np-close'
+      }
+    };
+
+    // First 'prime-the-pump' with an ActionURL
+    var initNotices = function($, settings, callback) {
+      $.ajax({
+        type:    'POST',
+        url:     settings.invokeNotificationServiceUrl,
+        async:   false,
+        success: function() {
+          fetchNotices($, settings, callback);
+        }
+      });
+    }
+
+    // Then fetch the notifications with a ResourceURL
+    var fetchNotices = function($, settings, callback) {
+      $.ajax({
+        url:      settings.getNotificationsUrl,
+        type:     'POST',
+        dataType: 'json',
+        success:  function (data) {
+          feed = data.feed;
+          callback(feed);
+        },
+        error:    function () {
+          container.html(" ").text("AJAX failed. ~ THE END ~");
+        }
+      });
+    }
+
+    upmodal_notice.launch = function ($, container, options) {
```

Comment:
:question: why is jQuery `$` being passed around as a param?

Reply:
Because it's not on the global namespace.  It needs to be provided to the `modal-notice.js`;  providing it this way retains some flexibility.

Reply:
(It needs to be up to the jsp to decide which instance of jQuery is used in the portlet.)

Reply:
Right that can also be achieved with

```js 
function ($) {
    /* scope with jQuery */
})(up.jQuery);
```